### PR TITLE
feature:support dynamic disable

### PR DESCRIPTION
**Pull request by slievrly**
### Ⅰ. Describe what this PR did
feature:support dynamic disable
1. If disable = true at startup, the runtime will always be disabled and cannot be changed dynamically;
2. If disable = false at startup, it can be changed at runtime

### CODE REVIEW

File: config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java [hunk 1]
```diff
@@ -131,40 +125,32 @@ public boolean removeConfig(String dataId, long timeoutMills) {
...
+        fileListener.onProcessEvent(new ConfigurationChangeEvent());
     }
 
     @Override
-    public void removeConfigListener(String dataId, ConfigChangeListener listener) {
-        List<ConfigChangeListener> configChangeListeners = getConfigListeners(dataId);
-        if (configChangeListeners == null) {
+    public void removeConfigListener(String dataId, ConfigurationChangeListener listener) {
+        Set<ConfigurationChangeListener> configChangeListeners = getConfigListeners(dataId);
+        if (dataId == null || configChangeListeners == null) {
```
Comment (xingfudeshi) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: listener==null

File: config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java [hunk 2]
```diff
@@ -128,53 +131,36 @@ public boolean removeConfig(String dataId, long timeoutMills) {
...
+
     }
 
     @Override
-    public void removeConfigListener(String dataId, ConfigChangeListener listener) {
-        List<ConfigChangeListener> configChangeListeners = getConfigListeners(dataId);
-        if (configChangeListeners == null) {
+    public void removeConfigListener(String dataId, ConfigurationChangeListener listener) {
+        Set<ConfigurationChangeListener> configChangeListeners = getConfigListeners(dataId);
+        if (configChangeListeners == null || listener == null) {
```
Comment (xingfudeshi) [inline config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java]: dataId==null

File: config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java [hunk 3]
```diff
@@ -128,53 +131,36 @@ public boolean removeConfig(String dataId, long timeoutMills) {
...
-        configChangeNotifiersMap.putIfAbsent(dataId, new ArrayList<>());
-        ConfigChangeNotifier configChangeNotifier = new ConfigChangeNotifier(dataId, listener);
-        configChangeNotifiersMap.get(dataId).add(configChangeNotifier);
-        if (null != listener.getExecutor()) {
-            listener.getExecutor().submit(configChangeNotifier);
-        } else {
-            consulNotifierExecutor.submit(configChangeNotifier);
-        }
+    public void addConfigListener(String dataId, ConfigurationChangeListener listener) {
+        configListenersMap.putIfAbsent(dataId, new ConcurrentSet<>());
```
Comment (xingfudeshi) [inline config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java]: dataId==null||listener==null

File: config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java [hunk 4]
```diff
@@ -131,40 +125,32 @@ public boolean removeConfig(String dataId, long timeoutMills) {
     }
 
     @Override
-    public void addConfigListener(String dataId, ConfigChangeListener listener) {
-        configListenersMap.putIfAbsent(dataId, new ArrayList<ConfigChangeListener>());
+    public void addConfigListener(String dataId, ConfigurationChangeListener listener) {
+        configListenersMap.putIfAbsent(dataId, new ConcurrentSet<>());
```
Comment (xingfudeshi) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: dataId==null||listener==null

**Comment by anrror** [comment]
When Client is restarted, The processing strategy of Server hasn't changed.

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 5]
```diff
@@ -170,4 +177,13 @@ private String formatMethod(Method method) {
         }
         return sb.append(")").toString();
     }
+
+    @Override
+    public void onChangeEvent(ConfigurationChangeEvent event) {
+        if (event.getDataId().equals(ConfigurationKeys.DISABLE_GLOBAL_TRANSACTION)) {
```
Comment (zjinlei) [inline spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: reverse equals

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 6]
```diff
@@ -170,4 +177,13 @@ private String formatMethod(Method method) {
...
+            LOGGER.info("{} config changed, old value:{}, new value:{}", ConfigurationKeys.DISABLE_GLOBAL_TRANSACTION,
+                event.getOldValue(), event.getNewValue());
```
Comment (zjinlei) [inline spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: event.getOldValue->disable

File: config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java (hunk 1 above)
Comment (slievrly) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: fixed.

File: config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java (hunk 2 above)
Comment (slievrly) [reply config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java]: fixed.

File: config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java (hunk 3 above)
Comment (slievrly) [reply config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java]: fixed.

File: config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java (hunk 4 above)
Comment (slievrly) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: fixed.

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java (hunk 5 above)
Comment (slievrly) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: fixed.

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java (hunk 6 above)
Comment (slievrly) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: fixed.

**Comment by objcoding** [comment]
file LGTM.

**Comment by anrror** [comment]
LGTM

**Comment by zjinlei** [comment]
@anrror Please note the configuration center you verified successfully.

**Comment by objcoding** [comment]
apollo LGTM.

**Comment by objcoding** [comment]
zk LGTM.

**Comment by objcoding** [comment]
consul LGTM.

**Comment by objcoding** [comment]
etcd3 LGTM.

**Pull request merged**